src/aktualizr_secondary/aktualizr_secondary.cc

```diff
@@ -79,23 +79,12 @@ void AktualizrSecondary::run() {
   // listen for messages
   std::shared_ptr<SecondaryPacket> pkt;
   while (channel_ >> pkt) {
-    std::cout << "Got packet " << pkt->str() << std::endl;
-
-    if (pkt->msg->variant == "Stop") {
-      // Will cause `accept()` to fail and break the loop
-      shutdown(*socket_hdl_, SHUT_RDWR);
-    }
+    // TODO: process message
   }
 
   tcp_thread.join();
 }
 
```

Comment:
It would be nice if we could keep the ability to stop the loop, it's useful for testing,
Perhaps, we can stop using a channel for now and move the tcp thread code directly in `run()`. `stop()` would be just a call to `shutdown()`

Reply:
Yes, I think it's better to dedicate channel to primary<->secondary communication and use something else to stop the loop. Alternatively we may have two different channels, one for communication and another for control (like 'event' and 'command' channels in primary).